sig-and-wg/sig-core/proposals/modularization.md

```diff
@@ -0,0 +1,72 @@
+- Add a new Remote Environment (also new type of application) in runtime.
+- Add completely new Kyma component that integrates with UI in runtime (register micro front-end and custom resources).
+- Use user provided S3 bucket instead of Minio.
+- Install private modules (from private Docker registry or private Git repository).
+- Install components from Kyma examples and showcases catalog easily (UI).
+
+## Current solution
+
+Current installation process is a mix of:
+
+- Custom shell scripts
+- Kubectl commands
+- Helm charts including one big, fat core chart
+- Complex parametrization
+
+Installer is a Kubernetes operator with hard-coded logic executing all the components above. Problems with this solution are as follows:
+
+- The default Minikube installation is slow and requires a lot of resources.
+- Shell scripts are difficult to manage (some of them have to support Mac, Linux, Windows).
+- New installation parameters or dependencies require code changes.
+
+## Proposal
+
+The proposed solution is to provide a flexible and modular installation of fine grained components with limited dependencies. The components installation and update should be more generic and declarative. This is the first set of ideas on how to improve the current solution:
+
+- Parameters should not be hard-coded - there should be a convention on how to pass the parameters through the installation process to the Kyma components.
+- Default values everywhere - the list of mandatory parameters should be as short as possible. The goal is to have only the cluster domain as a mandatory parameter.
+- Optional dependencies, for example: Kyma ui-api-layer can work without custom resources introduced by the Service Catalog or Remote Environment and still can support other areas.
+- Dynamic UI (micro front-ends) + back-end - adding new components extends views and also the backed functionalities, such as the supported GraphQL queries.
+- No custom logic in the installation process (shell scripts) - it should be replaced by the operator pattern.
+- The minimal set of mandatory components that form the real Kyma "core". The dependency for the core can be mandatory, other dependencies should be optional.
+- Kyma core as Gardener add-on. For reference, see [Knative with Gardener](https://github.com/knative/docs/blob/master/install/Knative-with-Gardener.md#alternative-way-to-enable-knative-with-gardener) and [Gardener Bouquet](https://github.com/gardener/bouquet) documentation.
+- Split Kyma core chart into more fine grained components, and provide a tooling to install all of them in the same convenient way as before (convention).
+- Kyma add-ons, brokers, and showcases as Kyma bundles. For reference, see the [Kyma bundles](https://github.com/PK85/kyma-bundles) repository.
+- Advanced Level: acceptance testing, especially ui-api-layer and cross components tests.
+
+### Modularity - refactoring
+
+![](assets/modularization.svg)
+
+### Kyma bundles for the Service Catalog
```

Comment:
This is one of sub-proposals and should not in my opinion be present in this proposal

Reply:
I'm going to extract that in a next phase, already decided that we will go with kyma bundles for brokers and showcases.

Reply:
maybe decided we do it, but proposal will explain how
anyway, not part of this PR ;)

Reply:
done